Source/Diagnostics/WarpXIO.cpp

```diff
@@ -746,29 +746,72 @@ WarpX::WriteJobInfo (const std::string& dir) const
 
 
 /* \brief
- *  The slice is ouput using visMF and can be visualized used amrvis. 
+ *  The raw slice data is written out in the plotfile format and can be visualized using yt. *  The slice data is written to diags/slice_plotfiles/pltXXXXX at the plotting interval.  
  */
 void
 WarpX::WriteSlicePlotFile () const
 {
-    if (F_fp[0] ) {
-       VisMF::Write( (*F_slice[0]), "vismf_F_slice");
-    }
+    // writing plotfile //
+    const std::string& slice_plotfilename = amrex::Concatenate(slice_plot_file,istep[0]);
+    amrex::Print() << " Writing slice plotfile " << slice_plotfilename << "\n";
 
-    if (rho_fp[0]) {
-       VisMF::Write( (*rho_slice[0]), "vismf_rho_slice");
-    }
+    const int ngrow = 0;
+
+    Vector<std::string> rfs;
+    VisMF::Header::Version current_version = VisMF::GetHeaderVersion();
+    VisMF::SetHeaderVersion(slice_plotfile_headerversion);
+    rfs.emplace_back("raw_fields");
+    
+    const int nlevels = finestLevel() + 1;
+     
+    // creating a temporary cell-centered dummy multifab //
+    // to get around the issue of yt complaining about no field data //
+    Vector< std::unique_ptr<MultiFab> > dummy_mf(nlevels);
+    const DistributionMapping &dm2 = Efield_slice[0][0]->DistributionMap();
+    Vector<std::string> varnames;
+    IntVect cc(AMREX_D_DECL(0,0,0));
+    for (int lev = 0; lev < nlevels; ++lev) 
+    {
+       dummy_mf[lev].reset(new MultiFab( 
+                     amrex::convert(Efield_slice[lev][0]->boxArray(),cc), 
+                     dm2, 1, 0 ));
+       dummy_mf[lev]->setVal(0.0);
+    } 
+    amrex::WriteMultiLevelPlotfile(slice_plotfilename, nlevels,
+                                   GetVecOfConstPtrs(dummy_mf), 
+                                   varnames, Geom(), t_new[0],
+                                   istep, refRatio(),
+                                   "HyperCLaw-V1.1", 
+                                   "Level_", "Cell", rfs);
+
+    for (int lev = 0; lev < nlevels; ++lev) 
+    {
+        const std::unique_ptr<MultiFab> empty_ptr;
+        const std::string raw_spltname = slice_plotfilename + "/raw_fields";
+        amrex::Print() << " raw spltname " << raw_spltname << "\n";
+        const DistributionMapping &dm = Efield_slice[lev][0]->DistributionMap();
+ 
+        WriteRawField( *Efield_slice[lev][0], dm, raw_spltname, level_prefix, "Ex_slice", lev, 0);
```

Comment:
Why is it required to write slice data as a raw fields?

Reply:
The slice data inherits the index type of the underlying data. 
As a result, we just write out the raw data without converting it to cell-centered values and then writing it as a plotfile.